docs/proposals/third-party-api-servers.md

```diff
@@ -0,0 +1,175 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/third-party-api-servers.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Third Party API Servers
+
+## Goal
+
+With third party API servers, we want to enable cluster admins or users to
```

Comment:
Hmm, the way I read your comment you're asking that we run a haproxy (or something) in front of each third party server. Isn't this a pain? 

This one paragraph is reminiscent of all the router/serviceloadbalancer stuff we did. Some services come up and register themselves with a single go program+haproxy capable of programming itself dynamically. I think this could be useful even today (without third party apiservers), because we'd just spin up 2 apiservers fronted by haproxy. 

We can give that router go program a "dumb" backend that doesn't need haproxy at all of course.